presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableQueries.java

```diff
@@ -0,0 +1,119 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createCustomer;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createLineitem;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createNationWithFormat;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createOrders;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createPart;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createPartSupp;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createRegion;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createSupplier;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.dropTableIfExists;
+import static com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils.createExternalTable;
+import static com.facebook.presto.nativeworker.SymlinkManifestGeneratorUtils.cleanupSymlinkData;
+import static com.facebook.presto.nativeworker.SymlinkManifestGeneratorUtils.getTpchTableColumns;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractTestNativeHiveExternalTableQueries
+        extends AbstractTestNativeTpchQueries
+{
+    private static final String HIVE = "hive";
+    private static final String TPCH = "tpch";
+    private static final String SYMLINK_FOLDER = "symlink_tables_manifests";
+    private static final String HIVE_DATA = "hive_data";
+
+    @Override
+    protected void createTables()
+    {
+        QueryRunner javaQueryRunner = (QueryRunner) getExpectedQueryRunner();
+        createOrders(javaQueryRunner);
+        createLineitem(javaQueryRunner);
+        createNationWithFormat(javaQueryRunner, "PARQUET");
+        createCustomer(javaQueryRunner);
+        createPart(javaQueryRunner);
+        createPartSupp(javaQueryRunner);
+        createRegion(javaQueryRunner);
+        createSupplier(javaQueryRunner);
+
+        createExternalTable(javaQueryRunner, TPCH, "orders", getTpchTableColumns("orders"));
+        createExternalTable(javaQueryRunner, TPCH, "lineitem", getTpchTableColumns("lineitem"));
+        createExternalTable(javaQueryRunner, TPCH, "nation", getTpchTableColumns("nation"));
+        createExternalTable(javaQueryRunner, TPCH, "customer", getTpchTableColumns("customer"));
+        createExternalTable(javaQueryRunner, TPCH, "part", getTpchTableColumns("part"));
+        createExternalTable(javaQueryRunner, TPCH, "partsupp", getTpchTableColumns("partsupp"));
+        createExternalTable(javaQueryRunner, TPCH, "region", getTpchTableColumns("region"));
+        createExternalTable(javaQueryRunner, TPCH, "supplier", getTpchTableColumns("supplier"));
+    }
+    @Test
+    public void testJoinQueries()
```

Comment:
You could remove testJoinQueries and testAggregationQueries. The tpch queries are sufficient.